onnxruntime/core/providers/qnn/builder/qnn_backend_manager.h

```diff
@@ -102,7 +103,10 @@ class QnnBackendManager {
 
   const QNN_INTERFACE_VER_TYPE& GetQnnInterface() { return qnn_interface_; }
 
-  const Qnn_ContextHandle_t& GetQnnContext() { return context_; }
+  const Qnn_ContextHandle_t& GetQnnContext(int index = 0) {
+    ORT_ENFORCE(contexts_.size() > 0, "No valid QNN context!");
```

Comment:
Maybe we can change this condition to `(index >= 0) && (index < contexts_.size())` to also catch an invalid index.